src/sentry/api/helpers/group_index/delete.py

```diff
@@ -110,6 +105,35 @@ def delete_group_list(
         )
 
 
+def delete_group_hashes(
+    project_id: int,
+    group_ids: Sequence[int],
+    seer_deletion: bool = False,
+) -> None:
+    hashes_batch_size = options.get("deletions.group-hashes-batch-size")
+    total_hashes = GroupHash.objects.filter(project_id=project_id, group_id__in=group_ids).count()
+
+    for _ in range(0, total_hashes, hashes_batch_size):
```

Comment:
I'd still suggest you add 10/20% to `total_hashes` here just in case the total number of `GroupHashes` somehow increases during the deletion process

Reply:
Another approach would be to `while True` and break out of that loop when you run out of `GroupHash` records in your read query. Because you're deleting in the `finally` block you should get to the end of the set eventually.